ext/phar/phar_object.c

```diff
@@ -482,15 +486,6 @@ PHP_METHOD(Phar, mount)
 carry_on:
 		if (SUCCESS != phar_mount_entry(pphar, actual, actual_len, path, path_len)) {
 			zend_throw_exception_ex(phar_ce_PharException, 0, "Mounting of %s to %s within phar %s failed", path, actual, arch);
-			if (path && path == entry) {
-				efree(entry);
-			}
-
-			if (arch) {
-				efree(arch);
-			}
-
-			goto finish;
```

Comment:
Why this is removed?

Reply:
The code outside the if branch is equivalent to this and performs the necessary clean-up and then jumps to ``finish``

Reply:
yeah, I see